src/rules/blas.jl

```diff
@@ -72,3 +74,37 @@ function rrule(f::typeof(BLAS.gemv), tA, A, x)
     Ω, (dtA, dα, dA, dx) = rrule(f, tA, one(eltype(A)), A, x)
     return Ω, (dtA, dA, dx)
 end
+
+#####
+##### `BLAS.gemm`
+#####
+
+function rrule(::typeof(gemm), tA::Char, tB::Char, α::Number,
+               A::AbstractMatrix, B::AbstractMatrix)
+    C = gemm(tA, tB, α, A, B)
+    #∂α = C̄ -> sum(C̄ .* C) / α  TODO: This is wrong and idk why
```

Comment:
Should this be
```julia
∂α = C̄ -> sum(C̄ .* C)
```

Reply:
Unfortunately that doesn't seem to do it:
```
gemm: Test Failed at /home/alex/.julia/dev/ChainRules/test/rules/blas.jl:22
  Expression: ≈(x̄_ad, x̄_fd, rtol=1.0e-9, atol=1.0e-9)
   Evaluated: 4.6207100689748355 ≈ -1.8353251336788157 (rtol=1.0e-9, atol=1.0e-9)
```

Reply:
Dividing by `α` is what Nabla does; I have no idea why it doesn't work in this instance.

Reply:
Hmm yeah, this is really weird. Frankly, I have no idea why Nabla does it...

Reply:
Perhaps we should add a `NotImplementedRule` and punt it down the road :stuck_out_tongue:

Reply:
> Frankly, I have no idea why Nabla does it

Tbh I don't either. I don't see how it would arise from the math.

Reply:
> Perhaps we should add a NotImplementedRule and punt it down the road 😛

I couldn't figure out how to deal with alpha so I actually did this in #28.